github-merge.py

```diff
@@ -42,6 +43,20 @@
     ATTR_WARN = '\033[1;31m'
     COMMIT_FORMAT = '%C(bold blue)%H%Creset %s %C(cyan)(%an)%Creset%C(green)%d%Creset'
 
+sanitize_mapping = {
+    # Disallow newlines
+    False: str.maketrans({x:None for x in range(0,32)}),
+    # Allow newlines
+    True: str.maketrans({x:None for x in range(0,32) if x != 10}),
+}
+def sanitize(s, newlines=True):
+    '''
+    Strip control characters (optionally except for newlines) from a string.
+    This prevent text data from doing potentially confusing or harmful things
+    with ANSI formatting, linefeeds bells etc.
+    '''
+    return s.translate(sanitize_mapping[newlines])
```

Comment:
I believe there are some higher unicode control characters as well, e.g. right-to-left text indicators, I don't know in what sense they affect the console.
But this gets the low-hanging fruit at least.